lib/middleware/authorize.js

```diff
@@ -0,0 +1,11 @@
+module.exports = async (req, res, next) => {
+  try {
+    if (!req.user || req.user.email !== 'admin')
```

Comment:
the way this is written combined with the way your data is structured, this just authorizes everyone who passes the authenticate middleware, or an admin, if you didn't have admin in your user table. if we look at authenticate, it checks for a cookie. ```jwt.verify()``` decodes the jwt using our ```JWT_SECRET``` and since the jwt stores our user object information, we can attach that user object to our request object, which is what happens on line 10 of your authenticate middleware. now, if we wanted to use our authorize middleware to check if a user has access to individual secrets, we would check that some aspect of our user object matches some aspect of our secret, but currently your if check won't throw an error as long as there is ANY user object attached from the authenticate middleware, so basically you're restricting access to allow any authenticated user, making authorize middleware superfluous